tests/mocha/blocks/procedures_test.js

```diff
@@ -394,13 +396,51 @@ suite('Procedures', function() {
           'Expected the second param input to exist');
         chai.assert.equal(
           callBlock.getFieldValue('ARGNAME0'),
-          'param1',
+          'param2',
           'Expected the first params field to match the name of the param');
         chai.assert.equal(
           callBlock.getFieldValue('ARGNAME1'),
-          'param2',
+          'param1',
           'Expected the second params field to match the name of the param');
       });
+
+      test(
+          'moving a parameter in the data model moves input blocks',
+          function() {
+            const defBlock = createProcDefBlock(this.workspace);
+            const callBlock = createProcCallBlock(this.workspace);
+            const procModel = defBlock.getProcedureModel();
+            const param1 =
+                new ObservableParameterModel(this.workspace, 'param1', 'id1');
+            const param2 =
+                new ObservableParameterModel(this.workspace, 'param2', 'id2');
+            procModel.insertParameter(param1, 0);
+            procModel.insertParameter(param2, 1);
+            const rowBlock1 = this.workspace.newBlock('row_block');
+            const rowBlock2 = this.workspace.newBlock('row_block');
+            callBlock.getInput('ARG0').connection
+                .connect(rowBlock1.outputConnection);
+            callBlock.getInput('ARG1').connection
+                .connect(rowBlock2.outputConnection);
+    
+            procModel.deleteParameter(1);
+            procModel.insertParameter(param2, 0);
+    
+            chai.assert.isNotNull(
+              callBlock.getInput('ARG0'),
+              'Expected the first param input to exist');
```

Comment:
Shouldn't these be indented more?

Reply:
Hmmm strange, I guess clang isn't running in our test directories.